src/FormattedMessage.js

```diff
@@ -0,0 +1,39 @@
+const format = (ctx, values) => ctx.parent.$formatMessage(ctx.props, values)
+const placeholder = (name) => `@\0@\0${name}\0@\0@`
+const placeholderRegex = /@\0@\0(.*?)\0@\0@/g
+
+export default {
+    functional: true,
+
+    props: {
+        id: {type: String, required: true},
+        defaultMessage: String,
+        values: Object,
+        tagName: {type: String, default: 'span'}
+    },
+
+    render (h, ctx) {
+        const slots = ctx.slots()
+        const slotNames = Object.keys(slots)
+        if (slotNames.length === 0) {
+            return h(ctx.props.tagName, format(ctx, ctx.props.values))
+        }
+
+        const values = Object.assign({}, ctx.props.values)
```

Comment:
I'm not sure I understand. How would the usage of this component look like? Not sure about the tendency your are referring to. Vue always references local component's "scope", not parent's. In case of functional, there is no instance, so the short-lived config is passed instead.